products/bash/release-products.sh

```diff
@@ -93,6 +93,94 @@ chart_repo=ibm-entitled-charts
 helm repo add ${chart_repo} https://raw.githubusercontent.com/IBM/charts/master/repo/entitled/
 
 
+# --- Tracing registration ---------------------------------------------
+# First parameter is the namespace where tracing is installed
+# Second parameter is the namespace to register. I.e. mq if registering mq
+# Note before calling this tracing should be installed and the app to be registered should be installed
+function register_tracing_application {
+  TRACING_NAMESPACE=${1}
+  REG_NAMESPACE=${2}
+
+  echo "Registering app in ${REG_NAMESPACE} namespace for tracing"
+
+  echo "Waiting for tracing registration job to complete..."
+  for i in `seq 1 60`; do
+  	TRACING_JOB_STATUS=$(kubectl get pods -n ${REG_NAMESPACE} | grep -E -m1 '(odtracing|od-registration)' | awk '{print $3}')
+  	if [ "$TRACING_JOB_STATUS" == "Completed" ]; then
+  		echo "Tracing registration job is complete."
+  		break
+  	else
+      if [[ i -eq 60 ]]; then
+        echo "Tracing registration job failed, giving up after numerous attempts"
+        exit 1
+      fi
+  		echo "Waiting for tracing registration job to complete (Attempt $i of 60). Job status: $TRACING_JOB_STATUS"
+  		kubectl -n ${REG_NAMESPACE} get pods,jobs
+  		echo "Checking again in one minute..."
+  		sleep 60
+  	fi
+  done
+
+  echo "Waiting 1 minute for tracing registration to trickle through"
+  sleep 60
+
+  echo "Waiting for tracing pod..."
+  # Loop/retry here, tracing pod may not exist yet
+  for i in `seq 1 60`; do
+    TRACING_POD=$(oc get pod -n ${TRACING_NAMESPACE} -l helm.sh/chart=ibm-icp4i-tracing-prod -o jsonpath='{.items[].metadata.name}')
+  	if [[ -z "$TRACING_POD" ]]; then
+      echo "Waiting for tracing pod (Attempt $i of 60), checking again in 15 seconds..."
+  		sleep 15
+  	else
+      echo "Got tracing pod: ${TRACING_POD}"
+  		break
+  	fi
+  done
+  if [[ -z "$TRACING_POD" ]]; then
+    echo "Failed to get tracing pod"
```

Comment:
should we fail here or should we ask the user to do the setup manually instead of exiting?

Reply:
If this tracing registration fails for APIC then APIC gateway will fail to start and the APIC configuration step won't work. I think it's hosed if this doesn't work, so just exit

Reply:
ok